src/provider-state/metamask.ts

```diff
@@ -0,0 +1,37 @@
+import { MetamaskProvider } from '../provider/types'
+
+export const MetamaskProviderStateSymbol = Symbol('MetamaskProviderState')
+
+export interface MetamaskProviderState {
+    [MetamaskProviderStateSymbol]: true
+    isMetaMask: true
+    autoRefreshOnNetworkChange: boolean
+    networkVersion: string
+    chainId: string
+    selectedAddress: string | null
+    accounts: string[]
+    isConnected: boolean
+    isEnabled: boolean
+    isApproved: Promise<boolean>
+    isUnlocked: Promise<boolean>
+}
+
+const getMetamaskProviderState = (provider: MetamaskProvider): MetamaskProviderState => {
+    const { isMetaMask, autoRefreshOnNetworkChange, networkVersion, chainId, selectedAddress } = provider
```

Comment:
is all this code backwards compatible for older Metamasks? Or should we version check somewhere

Reply:
No idea, honestly
I don't expect older metamasks to still be around. It's an extension, it's updated automatically for people
I'm more concerned about newer metamasks that may change things